src/py/flwr/common/telemetry.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2023 Adap GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower Telemetry."""
+
+import datetime
+import json
+import os
+import platform
+import urllib.request
+import uuid
+from enum import Enum, auto
+
+FLWR_TELEMETRY_ENABLED = os.getenv("FLWR_TELEMETRY_ENABLED", "1")
+FLWR_TELEMETRY_LOGGING = os.getenv("FLWR_TELEMETRY_LOGGING", "0")
+
+TELEMETRY_EVENTS_URL = "https://telemetry.flower.dev/api/v1/event"
+
+# A random session to identify, for example, on which platforms a START
+# event is emitted without a STOP event.
+SESSION = str(uuid.uuid4())
+
+# Using str as first base type to make it JSON serializable
+# otherwise we will get the following exception:
+# TypeError: Object of type EventType is not JSON serializable
+class EventType(str, Enum):
+    """Types of telemetry events."""
+
+    # Checkout Python docs for why this code is here:
+    # https://docs.python.org/3.7/library/enum.html#using-automatic-values
+    # Also this function signature is super weird the only reasonable option
+    # was to disable all checks for it.
+    def _generate_next_value_(name, start, count, last_values):  # type: ignore # pylint: disable=no-self-argument,arguments-differ # noqa: E501
+        return name
+
+    # Client
+    START_CLIENT = auto()
+    STOP_CLIENT = auto()
```

Comment:
How about:
```suggestion
    START_CLIENT_ENTER = auto()
    START_CLIENT_LEAVE = auto()
```
...this would make it more clear that both events are associated with the `start_client` function.

Reply:
(same for the other events below)